### ISSUE TITLE
Pulling changes into HA

### ISSUE THREAD

**Issue by mlowijs** [krocat/ToonHA#21]
Wasn't sure where to ask this, but here we go: at what point will the changes in this repository be merged into the HA repository? Are there rules for that?
I've been running the code in this repository for a while now and it seems to work fine.

Means I won't have to run custom_components anymore 😄

**Comment by boltgolt** [comment]
What i hear is that there's a new API coming, i think we should merge this again when that's tested and implemented. There are no real rules for that though, we could merge this at any time.

**Comment by mlowijs** [comment]
Okay! Good to know, and that sounds like a good idea. Surely many people will appreciated the improved stability of this code base over the older version. 😃

**Comment by costastf** [comment]
Not much more stability I am afraid since the api that feeds the data is
very unstable. I will soon look at the new api and see how that goes.

**Comment by costastf** [comment]
Just a quick update here, toonapilib is around 95% done. Currently missing the active part of actually setting values and switching things on and off and enabling the caching. Should be done the next time I sit with it. We can retain full compatibility with the old toonlib so the only thing that we will need once the library is done is to update the component for the new extra arguments of oath_secret and oauth_key and the documentation on how people should register on the new api and get those keys to setup their components. Hoping to be done with the library by the end of the year. Cheers!

**Comment by mlowijs** [comment]
Great work @costastf! 😀

**Comment by costastf** [comment]
The discussion about this issue is happening at https://community.home-assistant.io/t/toon-van-eneco-custom-component/16027/188 so I am closing this.
